{"metadata":{"kernelspec":{"language":"python","display_name":"Python 3","name":"python3"},"language_info":{"name":"python","version":"3.10.12","mimetype":"text/x-python","codemirror_mode":{"name":"ipython","version":3},"pygments_lexer":"ipython3","nbconvert_exporter":"python","file_extension":".py"},"kaggle":{"accelerator":"nvidiaTeslaT4","dataSources":[{"sourceId":10788747,"sourceType":"datasetVersion","datasetId":6695172},{"sourceId":10788812,"sourceType":"datasetVersion","datasetId":6695220},{"sourceId":223293795,"sourceType":"kernelVersion"},{"sourceId":223293856,"sourceType":"kernelVersion"}],"dockerImageVersionId":30886,"isInternetEnabled":true,"language":"python","sourceType":"notebook","isGpuEnabled":true}},"nbformat_minor":4,"nbformat":4,"cells":[{"cell_type":"code","source":"# This Python 3 environment comes with many helpful analytics libraries installed\n# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python\n# For example, here's several helpful packages to load\n\nimport numpy as np # linear algebra\nimport pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)\n\n# Input data files are available in the read-only \"../input/\" directory\n# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory\n\nimport os\nfor dirname, _, filenames in os.walk('/kaggle/input'):\n    for filename in filenames:\n        print(os.path.join(dirname, filename))\n\n# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using \"Save & Run All\" \n# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session","metadata":{"_uuid":"8f2839f25d086af736a60e9eeb907d3b93b6e0e5","_cell_guid":"b1076dfc-b9ad-4769-8c92-a6c4dae69d19","trusted":true,"execution":{"iopub.status.busy":"2025-02-19T03:19:18.183247Z","iopub.execute_input":"2025-02-19T03:19:18.183531Z","iopub.status.idle":"2025-02-19T03:19:19.064392Z","shell.execute_reply.started":"2025-02-19T03:19:18.183508Z","shell.execute_reply":"2025-02-19T03:19:19.063664Z"}},"outputs":[{"name":"stdout","text":"/kaggle/input/deepseek-rec-data/train_set(2).json\n/kaggle/input/deepseek-rec-data/test_set.json\n/kaggle/input/deepspeed-config/ds_config_zero3.json\n","output_type":"stream"}],"execution_count":2},{"cell_type":"code","source":"train_data = pd.read_json(\"/kaggle/input/deepseek-rec-data/train_set(2).json\")\ntest_data = pd.read_json(\"/kaggle/input/deepseek-rec-data/test_set.json\")\n\nds_config = pd.read_json(\"/kaggle/input/deepspeed-config/ds_config_zero3.json\")\n\noutput_path = \"results/\"","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-02-19T03:19:19.573964Z","iopub.execute_input":"2025-02-19T03:19:19.574349Z","iopub.status.idle":"2025-02-19T03:19:19.665435Z","shell.execute_reply.started":"2025-02-19T03:19:19.574321Z","shell.execute_reply":"2025-02-19T03:19:19.664800Z"}},"outputs":[],"execution_count":3},{"cell_type":"code","source":"!pip install torch==2.0.1 tokenizers==0.14.0 transformers==4.35.0 accelerate==0.24.1 attrdict tqdm deepspeed==0.12.2 datasets tensorboardX\n","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-02-19T03:34:52.461635Z","iopub.execute_input":"2025-02-19T03:34:52.461940Z","iopub.status.idle":"2025-02-19T03:35:07.191977Z","shell.execute_reply.started":"2025-02-19T03:34:52.461914Z","shell.execute_reply":"2025-02-19T03:35:07.191144Z"}},"outputs":[{"name":"stdout","text":"Requirement already satisfied: torch==2.0.1 in /usr/local/lib/python3.10/dist-packages (2.0.1)\nRequirement already satisfied: tokenizers==0.14.0 in /usr/local/lib/python3.10/dist-packages (0.14.0)\nCollecting transformers==4.35.0\n  Downloading transformers-4.35.0-py3-none-any.whl.metadata (123 kB)\n\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m123.1/123.1 kB\u001b[0m \u001b[31m3.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hCollecting accelerate==0.24.1\n  Downloading accelerate-0.24.1-py3-none-any.whl.metadata (18 kB)\nCollecting attrdict\n  Downloading attrdict-2.0.1-py2.py3-none-any.whl.metadata (6.7 kB)\nRequirement already satisfied: tqdm in /usr/local/lib/python3.10/dist-packages (4.67.1)\nRequirement already satisfied: deepspeed==0.12.2 in /usr/local/lib/python3.10/dist-packages (0.12.2)\nRequirement already satisfied: datasets in /usr/local/lib/python3.10/dist-packages (3.2.0)\nCollecting tensorboardX\n  Downloading tensorboardX-2.6.2.2-py2.py3-none-any.whl.metadata (5.8 kB)\nRequirement already satisfied: filelock in /usr/local/lib/python3.10/dist-packages (from torch==2.0.1) (3.17.0)\nRequirement already satisfied: typing-extensions in /usr/local/lib/python3.10/dist-packages (from torch==2.0.1) (4.12.2)\nRequirement already satisfied: sympy in /usr/local/lib/python3.10/dist-packages (from torch==2.0.1) (1.13.1)\nRequirement already satisfied: networkx in /usr/local/lib/python3.10/dist-packages (from torch==2.0.1) (3.4.2)\nRequirement already satisfied: jinja2 in /usr/local/lib/python3.10/dist-packages (from torch==2.0.1) (3.1.4)\nRequirement already satisfied: nvidia-cuda-nvrtc-cu11==11.7.99 in /usr/local/lib/python3.10/dist-packages (from torch==2.0.1) (11.7.99)\nRequirement already satisfied: nvidia-cuda-runtime-cu11==11.7.99 in /usr/local/lib/python3.10/dist-packages (from torch==2.0.1) (11.7.99)\nRequirement already satisfied: nvidia-cuda-cupti-cu11==11.7.101 in /usr/local/lib/python3.10/dist-packages (from torch==2.0.1) (11.7.101)\nRequirement already satisfied: nvidia-cudnn-cu11==8.5.0.96 in /usr/local/lib/python3.10/dist-packages (from torch==2.0.1) (8.5.0.96)\nRequirement already satisfied: nvidia-cublas-cu11==11.10.3.66 in /usr/local/lib/python3.10/dist-packages (from torch==2.0.1) (11.10.3.66)\nRequirement already satisfied: nvidia-cufft-cu11==10.9.0.58 in /usr/local/lib/python3.10/dist-packages (from torch==2.0.1) (10.9.0.58)\nRequirement already satisfied: nvidia-curand-cu11==10.2.10.91 in /usr/local/lib/python3.10/dist-packages (from torch==2.0.1) (10.2.10.91)\nRequirement already satisfied: nvidia-cusolver-cu11==11.4.0.1 in /usr/local/lib/python3.10/dist-packages (from torch==2.0.1) (11.4.0.1)\nRequirement already satisfied: nvidia-cusparse-cu11==11.7.4.91 in /usr/local/lib/python3.10/dist-packages (from torch==2.0.1) (11.7.4.91)\nRequirement already satisfied: nvidia-nccl-cu11==2.14.3 in /usr/local/lib/python3.10/dist-packages (from torch==2.0.1) (2.14.3)\nRequirement already satisfied: nvidia-nvtx-cu11==11.7.91 in /usr/local/lib/python3.10/dist-packages (from torch==2.0.1) (11.7.91)\nRequirement already satisfied: triton==2.0.0 in /usr/local/lib/python3.10/dist-packages (from torch==2.0.1) (2.0.0)\nRequirement already satisfied: huggingface_hub<0.17,>=0.16.4 in /usr/local/lib/python3.10/dist-packages (from tokenizers==0.14.0) (0.16.4)\nRequirement already satisfied: numpy>=1.17 in /usr/local/lib/python3.10/dist-packages (from transformers==4.35.0) (1.26.4)\nRequirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.10/dist-packages (from transformers==4.35.0) (24.2)\nRequirement already satisfied: pyyaml>=5.1 in /usr/local/lib/python3.10/dist-packages (from transformers==4.35.0) (6.0.2)\nRequirement already satisfied: regex!=2019.12.17 in /usr/local/lib/python3.10/dist-packages (from transformers==4.35.0) (2024.11.6)\nRequirement already satisfied: requests in /usr/local/lib/python3.10/dist-packages (from transformers==4.35.0) (2.32.3)\nRequirement already satisfied: safetensors>=0.3.1 in /usr/local/lib/python3.10/dist-packages (from transformers==4.35.0) (0.4.5)\nRequirement already satisfied: psutil in /usr/local/lib/python3.10/dist-packages (from accelerate==0.24.1) (5.9.5)\nRequirement already satisfied: hjson in /usr/local/lib/python3.10/dist-packages (from deepspeed==0.12.2) (3.1.0)\nRequirement already satisfied: ninja in /usr/local/lib/python3.10/dist-packages (from deepspeed==0.12.2) (1.11.1.3)\nRequirement already satisfied: py-cpuinfo in /usr/local/lib/python3.10/dist-packages (from deepspeed==0.12.2) (9.0.0)\nRequirement already satisfied: pydantic in /usr/local/lib/python3.10/dist-packages (from deepspeed==0.12.2) (2.11.0a1)\nRequirement already satisfied: pynvml in /usr/local/lib/python3.10/dist-packages (from deepspeed==0.12.2) (11.4.1)\nRequirement already satisfied: setuptools in /usr/local/lib/python3.10/dist-packages (from nvidia-cublas-cu11==11.10.3.66->torch==2.0.1) (75.1.0)\nRequirement already satisfied: wheel in /usr/local/lib/python3.10/dist-packages (from nvidia-cublas-cu11==11.10.3.66->torch==2.0.1) (0.45.1)\nRequirement already satisfied: cmake in /usr/local/lib/python3.10/dist-packages (from triton==2.0.0->torch==2.0.1) (3.31.2)\nRequirement already satisfied: lit in /usr/local/lib/python3.10/dist-packages (from triton==2.0.0->torch==2.0.1) (18.1.8)\nRequirement already satisfied: six in /usr/local/lib/python3.10/dist-packages (from attrdict) (1.17.0)\nRequirement already satisfied: pyarrow>=15.0.0 in /usr/local/lib/python3.10/dist-packages (from datasets) (19.0.0)\nRequirement already satisfied: dill<0.3.9,>=0.3.0 in /usr/local/lib/python3.10/dist-packages (from datasets) (0.3.8)\nRequirement already satisfied: pandas in /usr/local/lib/python3.10/dist-packages (from datasets) (2.2.3)\nRequirement already satisfied: xxhash in /usr/local/lib/python3.10/dist-packages (from datasets) (3.5.0)\nRequirement already satisfied: multiprocess<0.70.17 in /usr/local/lib/python3.10/dist-packages (from datasets) (0.70.16)\nRequirement already satisfied: fsspec<=2024.9.0,>=2023.1.0 in /usr/local/lib/python3.10/dist-packages (from fsspec[http]<=2024.9.0,>=2023.1.0->datasets) (2024.9.0)\nRequirement already satisfied: aiohttp in /usr/local/lib/python3.10/dist-packages (from datasets) (3.11.11)\nINFO: pip is looking at multiple versions of datasets to determine which version is compatible with other requirements. This could take a while.\nCollecting datasets\n  Downloading datasets-3.3.1-py3-none-any.whl.metadata (19 kB)\n  Downloading datasets-3.3.0-py3-none-any.whl.metadata (19 kB)\n  Downloading datasets-3.1.0-py3-none-any.whl.metadata (20 kB)\n  Downloading datasets-3.0.2-py3-none-any.whl.metadata (20 kB)\n  Downloading datasets-3.0.1-py3-none-any.whl.metadata (20 kB)\nCollecting fsspec<=2024.6.1,>=2023.1.0 (from fsspec[http]<=2024.6.1,>=2023.1.0->datasets)\n  Downloading fsspec-2024.6.1-py3-none-any.whl.metadata (11 kB)\nCollecting datasets\n  Downloading datasets-3.0.0-py3-none-any.whl.metadata (19 kB)\n  Downloading datasets-2.21.0-py3-none-any.whl.metadata (21 kB)\nINFO: pip is still looking at multiple versions of datasets to determine which version is compatible with other requirements. This could take a while.\n  Downloading datasets-2.20.0-py3-none-any.whl.metadata (19 kB)\nCollecting pyarrow-hotfix (from datasets)\n  Downloading pyarrow_hotfix-0.6-py3-none-any.whl.metadata (3.6 kB)\nCollecting fsspec<=2024.5.0,>=2023.1.0 (from fsspec[http]<=2024.5.0,>=2023.1.0->datasets)\n  Downloading fsspec-2024.5.0-py3-none-any.whl.metadata (11 kB)\nCollecting datasets\n  Downloading datasets-2.19.2-py3-none-any.whl.metadata (19 kB)\nCollecting fsspec<=2024.3.1,>=2023.1.0 (from fsspec[http]<=2024.3.1,>=2023.1.0->datasets)\n  Downloading fsspec-2024.3.1-py3-none-any.whl.metadata (6.8 kB)\nCollecting datasets\n  Downloading datasets-2.19.1-py3-none-any.whl.metadata (19 kB)\n  Downloading datasets-2.19.0-py3-none-any.whl.metadata (19 kB)\n  Downloading datasets-2.18.0-py3-none-any.whl.metadata (20 kB)\nCollecting fsspec<=2024.2.0,>=2023.1.0 (from fsspec[http]<=2024.2.0,>=2023.1.0->datasets)\n  Downloading fsspec-2024.2.0-py3-none-any.whl.metadata (6.8 kB)\nINFO: This is taking longer than usual. You might need to provide the dependency resolver with stricter constraints to reduce runtime. See https://pip.pypa.io/warnings/backtracking for guidance. If you want to abort this run, press Ctrl + C.\nCollecting datasets\n  Downloading datasets-2.17.1-py3-none-any.whl.metadata (20 kB)\nCollecting fsspec<=2023.10.0,>=2023.1.0 (from fsspec[http]<=2023.10.0,>=2023.1.0->datasets)\n  Downloading fsspec-2023.10.0-py3-none-any.whl.metadata (6.8 kB)\nCollecting datasets\n  Downloading datasets-2.17.0-py3-none-any.whl.metadata (20 kB)\n  Downloading datasets-2.16.1-py3-none-any.whl.metadata (20 kB)\nCollecting dill<0.3.8,>=0.3.0 (from datasets)\n  Downloading dill-0.3.7-py3-none-any.whl.metadata (9.9 kB)\nCollecting datasets\n  Downloading datasets-2.16.0-py3-none-any.whl.metadata (20 kB)\n  Downloading datasets-2.15.0-py3-none-any.whl.metadata (20 kB)\n  Downloading datasets-2.14.7-py3-none-any.whl.metadata (19 kB)\nRequirement already satisfied: protobuf>=3.20 in /usr/local/lib/python3.10/dist-packages (from tensorboardX) (3.20.3)\nRequirement already satisfied: aiohappyeyeballs>=2.3.0 in /usr/local/lib/python3.10/dist-packages (from aiohttp->datasets) (2.4.4)\nRequirement already satisfied: aiosignal>=1.1.2 in /usr/local/lib/python3.10/dist-packages (from aiohttp->datasets) (1.3.2)\nRequirement already satisfied: async-timeout<6.0,>=4.0 in /usr/local/lib/python3.10/dist-packages (from aiohttp->datasets) (5.0.1)\nRequirement already satisfied: attrs>=17.3.0 in /usr/local/lib/python3.10/dist-packages (from aiohttp->datasets) (25.1.0)\nRequirement already satisfied: frozenlist>=1.1.1 in /usr/local/lib/python3.10/dist-packages (from aiohttp->datasets) (1.5.0)\nRequirement already satisfied: multidict<7.0,>=4.5 in /usr/local/lib/python3.10/dist-packages (from aiohttp->datasets) (6.1.0)\nRequirement already satisfied: propcache>=0.2.0 in /usr/local/lib/python3.10/dist-packages (from aiohttp->datasets) (0.2.1)\nRequirement already satisfied: yarl<2.0,>=1.17.0 in /usr/local/lib/python3.10/dist-packages (from aiohttp->datasets) (1.18.3)\nRequirement already satisfied: mkl_fft in /usr/local/lib/python3.10/dist-packages (from numpy>=1.17->transformers==4.35.0) (1.3.8)\nRequirement already satisfied: mkl_random in /usr/local/lib/python3.10/dist-packages (from numpy>=1.17->transformers==4.35.0) (1.2.4)\nRequirement already satisfied: mkl_umath in /usr/local/lib/python3.10/dist-packages (from numpy>=1.17->transformers==4.35.0) (0.1.1)\nRequirement already satisfied: mkl in /usr/local/lib/python3.10/dist-packages (from numpy>=1.17->transformers==4.35.0) (2025.0.1)\nRequirement already satisfied: tbb4py in /usr/local/lib/python3.10/dist-packages (from numpy>=1.17->transformers==4.35.0) (2022.0.0)\nRequirement already satisfied: mkl-service in /usr/local/lib/python3.10/dist-packages (from numpy>=1.17->transformers==4.35.0) (2.4.1)\nRequirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.10/dist-packages (from requests->transformers==4.35.0) (3.4.1)\nRequirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.10/dist-packages (from requests->transformers==4.35.0) (3.10)\nRequirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.10/dist-packages (from requests->transformers==4.35.0) (2.3.0)\nRequirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.10/dist-packages (from requests->transformers==4.35.0) (2025.1.31)\nRequirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.10/dist-packages (from jinja2->torch==2.0.1) (3.0.2)\nINFO: pip is looking at multiple versions of multiprocess to determine which version is compatible with other requirements. This could take a while.\nCollecting multiprocess (from datasets)\n  Downloading multiprocess-0.70.17-py310-none-any.whl.metadata (7.2 kB)\n  Downloading multiprocess-0.70.15-py310-none-any.whl.metadata (7.2 kB)\nRequirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.10/dist-packages (from pandas->datasets) (2.9.0.post0)\nRequirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.10/dist-packages (from pandas->datasets) (2025.1)\nRequirement already satisfied: tzdata>=2022.7 in /usr/local/lib/python3.10/dist-packages (from pandas->datasets) (2025.1)\nRequirement already satisfied: annotated-types>=0.6.0 in /usr/local/lib/python3.10/dist-packages (from pydantic->deepspeed==0.12.2) (0.7.0)\nRequirement already satisfied: pydantic-core==2.28.0 in /usr/local/lib/python3.10/dist-packages (from pydantic->deepspeed==0.12.2) (2.28.0)\nRequirement already satisfied: mpmath<1.4,>=1.1.0 in /usr/local/lib/python3.10/dist-packages (from sympy->torch==2.0.1) (1.3.0)\nRequirement already satisfied: intel-openmp>=2024 in /usr/local/lib/python3.10/dist-packages (from mkl->numpy>=1.17->transformers==4.35.0) (2024.2.0)\nRequirement already satisfied: tbb==2022.* in /usr/local/lib/python3.10/dist-packages (from mkl->numpy>=1.17->transformers==4.35.0) (2022.0.0)\nRequirement already satisfied: tcmlib==1.* in /usr/local/lib/python3.10/dist-packages (from tbb==2022.*->mkl->numpy>=1.17->transformers==4.35.0) (1.2.0)\nRequirement already satisfied: intel-cmplr-lib-rt in /usr/local/lib/python3.10/dist-packages (from mkl_umath->numpy>=1.17->transformers==4.35.0) (2024.2.0)\nRequirement already satisfied: intel-cmplr-lib-ur==2024.2.0 in /usr/local/lib/python3.10/dist-packages (from intel-openmp>=2024->mkl->numpy>=1.17->transformers==4.35.0) (2024.2.0)\nDownloading transformers-4.35.0-py3-none-any.whl (7.9 MB)\n\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m7.9/7.9 MB\u001b[0m \u001b[31m63.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m00:01\u001b[0m00:01\u001b[0m\n\u001b[?25hDownloading accelerate-0.24.1-py3-none-any.whl (261 kB)\n\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m261.4/261.4 kB\u001b[0m \u001b[31m19.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hDownloading attrdict-2.0.1-py2.py3-none-any.whl (9.9 kB)\nDownloading datasets-2.14.7-py3-none-any.whl (520 kB)\n\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m520.4/520.4 kB\u001b[0m \u001b[31m34.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hDownloading tensorboardX-2.6.2.2-py2.py3-none-any.whl (101 kB)\n\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m101.7/101.7 kB\u001b[0m \u001b[31m8.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hDownloading dill-0.3.7-py3-none-any.whl (115 kB)\n\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m115.3/115.3 kB\u001b[0m \u001b[31m9.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hDownloading fsspec-2023.10.0-py3-none-any.whl (166 kB)\n\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m166.4/166.4 kB\u001b[0m \u001b[31m12.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hDownloading multiprocess-0.70.15-py310-none-any.whl (134 kB)\n\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m134.8/134.8 kB\u001b[0m \u001b[31m11.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hDownloading pyarrow_hotfix-0.6-py3-none-any.whl (7.9 kB)\nInstalling collected packages: pyarrow-hotfix, fsspec, dill, attrdict, multiprocess, transformers, tensorboardX, datasets, accelerate\n  Attempting uninstall: fsspec\n    Found existing installation: fsspec 2024.9.0\n    Uninstalling fsspec-2024.9.0:\n      Successfully uninstalled fsspec-2024.9.0\n  Attempting uninstall: dill\n    Found existing installation: dill 0.3.8\n    Uninstalling dill-0.3.8:\n      Successfully uninstalled dill-0.3.8\n  Attempting uninstall: multiprocess\n    Found existing installation: multiprocess 0.70.16\n    Uninstalling multiprocess-0.70.16:\n      Successfully uninstalled multiprocess-0.70.16\n  Attempting uninstall: transformers\n    Found existing installation: transformers 4.47.0\n    Uninstalling transformers-4.47.0:\n      Successfully uninstalled transformers-4.47.0\n  Attempting uninstall: datasets\n    Found existing installation: datasets 3.2.0\n    Uninstalling datasets-3.2.0:\n      Successfully uninstalled datasets-3.2.0\n  Attempting uninstall: accelerate\n    Found existing installation: accelerate 1.2.1\n    Uninstalling accelerate-1.2.1:\n      Successfully uninstalled accelerate-1.2.1\n\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\ncudf-cu12 24.12.0 requires pyarrow<19.0.0a0,>=14.0.0; platform_machine == \"x86_64\", but you have pyarrow 19.0.0 which is incompatible.\ngcsfs 2024.10.0 requires fsspec==2024.10.0, but you have fsspec 2023.10.0 which is incompatible.\npathos 0.3.2 requires dill>=0.3.8, but you have dill 0.3.7 which is incompatible.\npathos 0.3.2 requires multiprocess>=0.70.16, but you have multiprocess 0.70.15 which is incompatible.\npeft 0.14.0 requires huggingface-hub>=0.25.0, but you have huggingface-hub 0.16.4 which is incompatible.\npytorch-lightning 2.5.0.post0 requires torch>=2.1.0, but you have torch 2.0.1 which is incompatible.\nsentence-transformers 3.3.1 requires huggingface-hub>=0.20.0, but you have huggingface-hub 0.16.4 which is incompatible.\nsentence-transformers 3.3.1 requires transformers<5.0.0,>=4.41.0, but you have transformers 4.35.0 which is incompatible.\u001b[0m\u001b[31m\n\u001b[0mSuccessfully installed accelerate-0.24.1 attrdict-2.0.1 datasets-2.14.7 dill-0.3.7 fsspec-2023.10.0 multiprocess-0.70.15 pyarrow-hotfix-0.6 tensorboardX-2.6.2.2 transformers-4.35.0\n","output_type":"stream"}],"execution_count":18},{"cell_type":"code","source":"!DATA_PATH=\"/kaggle/input/deepseek-rec-data/train_set(2).json\"\n!OUTPUT_PATH=\"/kaggle/working/results/\"\n!MODEL_PATH=\"deepseek-ai/deepseek-coder-6.7b-instruct\"\n\n!deepspeed /kaggle/usr/lib/deepseek_tune/finetune_deepseekcoder.py \\\n    --model_name_or_path $MODEL_PATH \\\n    --data_path $DATA_PATH \\\n    --output_dir $OUTPUT_PATH \\\n    --num_train_epochs 3 \\\n    --model_max_length 1024 \\\n    --per_device_train_batch_size 16 \\\n    --per_device_eval_batch_size 1 \\\n    --gradient_accumulation_steps 4 \\\n    --evaluation_strategy \"no\" \\\n    --save_strategy \"steps\" \\\n    --save_steps 100 \\\n    --save_total_limit 100 \\\n    --learning_rate 2e-5 \\\n    --warmup_steps 10 \\\n    --logging_steps 1 \\\n    --lr_scheduler_type \"cosine\" \\\n    --gradient_checkpointing True \\\n    --report_to \"tensorboard\" \\\n    --deepspeed /kaggle/input/deepspeed-config/ds_config_zero3.json \\\n    --bf16 True","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-02-19T03:35:07.193162Z","iopub.execute_input":"2025-02-19T03:35:07.193461Z","iopub.status.idle":"2025-02-19T03:35:39.655712Z","shell.execute_reply.started":"2025-02-19T03:35:07.193422Z","shell.execute_reply":"2025-02-19T03:35:39.654818Z"}},"outputs":[{"name":"stdout","text":"[2025-02-19 03:35:11,358] [INFO] [real_accelerator.py:158:get_accelerator] Setting ds_accelerator to cuda (auto detect)\n[2025-02-19 03:35:17,559] [WARNING] [runner.py:203:fetch_hostfile] Unable to find hostfile, will proceed with training with local resources only.\n[2025-02-19 03:35:17,559] [INFO] [runner.py:570:main] cmd = /usr/bin/python3 -u -m deepspeed.launcher.launch --world_info=eyJsb2NhbGhvc3QiOiBbMCwgMV19 --master_addr=127.0.0.1 --master_port=29500 --enable_each_rank_log=None /kaggle/usr/lib/deepseek_tune/finetune_deepseekcoder.py --model_name_or_path --data_path --output_dir --num_train_epochs 3 --model_max_length 1024 --per_device_train_batch_size 16 --per_device_eval_batch_size 1 --gradient_accumulation_steps 4 --evaluation_strategy no --save_strategy steps --save_steps 100 --save_total_limit 100 --learning_rate 2e-5 --warmup_steps 10 --logging_steps 1 --lr_scheduler_type cosine --gradient_checkpointing True --report_to tensorboard --deepspeed /kaggle/input/deepspeed-config/ds_config_zero3.json --bf16 True\n[2025-02-19 03:35:19,641] [INFO] [real_accelerator.py:158:get_accelerator] Setting ds_accelerator to cuda (auto detect)\n[2025-02-19 03:35:23,012] [INFO] [launch.py:138:main] 0 NV_LIBNCCL_DEV_PACKAGE=libnccl-dev=2.19.3-1+cuda12.2\n[2025-02-19 03:35:23,012] [INFO] [launch.py:138:main] 0 NV_LIBNCCL_DEV_PACKAGE_VERSION=2.19.3-1\n[2025-02-19 03:35:23,012] [INFO] [launch.py:138:main] 0 NCCL_VERSION=2.19.3-1\n[2025-02-19 03:35:23,012] [INFO] [launch.py:138:main] 0 NV_LIBNCCL_DEV_PACKAGE_NAME=libnccl-dev\n[2025-02-19 03:35:23,012] [INFO] [launch.py:138:main] 0 NV_LIBNCCL_PACKAGE=libnccl2=2.19.3-1+cuda12.2\n[2025-02-19 03:35:23,013] [INFO] [launch.py:138:main] 0 NV_LIBNCCL_PACKAGE_NAME=libnccl2\n[2025-02-19 03:35:23,013] [INFO] [launch.py:138:main] 0 NV_LIBNCCL_PACKAGE_VERSION=2.19.3-1\n[2025-02-19 03:35:23,013] [INFO] [launch.py:145:main] WORLD INFO DICT: {'localhost': [0, 1]}\n[2025-02-19 03:35:23,013] [INFO] [launch.py:151:main] nnodes=1, num_local_procs=2, node_rank=0\n[2025-02-19 03:35:23,013] [INFO] [launch.py:162:main] global_rank_mapping=defaultdict(<class 'list'>, {'localhost': [0, 1]})\n[2025-02-19 03:35:23,013] [INFO] [launch.py:163:main] dist_world_size=2\n[2025-02-19 03:35:23,013] [INFO] [launch.py:165:main] Setting CUDA_VISIBLE_DEVICES=0,1\n2025-02-19 03:35:27.304960: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:485] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered\n2025-02-19 03:35:27.304960: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:485] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered\n2025-02-19 03:35:27.500088: E external/local_xla/xla/stream_executor/cuda/cuda_dnn.cc:8454] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered\n2025-02-19 03:35:27.500103: E external/local_xla/xla/stream_executor/cuda/cuda_dnn.cc:8454] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered\n2025-02-19 03:35:27.557692: E external/local_xla/xla/stream_executor/cuda/cuda_blas.cc:1452] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered\n2025-02-19 03:35:27.557703: E external/local_xla/xla/stream_executor/cuda/cuda_blas.cc:1452] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.10/dist-packages/transformers/utils/import_utils.py\", line 1345, in _get_module\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.10/dist-packages/transformers/utils/import_utils.py\", line 1345, in _get_module\n    return importlib.import_module(\".\" + module_name, self.__name__)\n  File \"/usr/lib/python3.10/importlib/__init__.py\", line 126, in import_module\n    return importlib.import_module(\".\" + module_name, self.__name__)\n  File \"/usr/lib/python3.10/importlib/__init__.py\", line 126, in import_module\n        return _bootstrap._gcd_import(name[level:], package, level)return _bootstrap._gcd_import(name[level:], package, level)\n\n  File \"<frozen importlib._bootstrap>\", line 1050, in _gcd_import\n  File \"<frozen importlib._bootstrap>\", line 1050, in _gcd_import\n  File \"<frozen importlib._bootstrap>\", line 1027, in _find_and_load\n  File \"<frozen importlib._bootstrap>\", line 1027, in _find_and_load\n  File \"<frozen importlib._bootstrap>\", line 1006, in _find_and_load_unlocked\n  File \"<frozen importlib._bootstrap>\", line 1006, in _find_and_load_unlocked\n  File \"<frozen importlib._bootstrap>\", line 688, in _load_unlocked\n  File \"<frozen importlib._bootstrap>\", line 688, in _load_unlocked\n  File \"<frozen importlib._bootstrap_external>\", line 883, in exec_module\n  File \"<frozen importlib._bootstrap_external>\", line 883, in exec_module\n  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n  File \"/usr/local/lib/python3.10/dist-packages/transformers/trainer.py\", line 190, in <module>\n  File \"/usr/local/lib/python3.10/dist-packages/transformers/trainer.py\", line 190, in <module>\n        from peft import PeftModelfrom peft import PeftModel\n\n  File \"/usr/local/lib/python3.10/dist-packages/peft/__init__.py\", line 22, in <module>\n  File \"/usr/local/lib/python3.10/dist-packages/peft/__init__.py\", line 22, in <module>\n    from .auto import (\n    from .auto import (  File \"/usr/local/lib/python3.10/dist-packages/peft/auto.py\", line 31, in <module>\n\n  File \"/usr/local/lib/python3.10/dist-packages/peft/auto.py\", line 31, in <module>\n    from .config import PeftConfig\n      File \"/usr/local/lib/python3.10/dist-packages/peft/config.py\", line 24, in <module>\nfrom .config import PeftConfig\n  File \"/usr/local/lib/python3.10/dist-packages/peft/config.py\", line 24, in <module>\n    from .utils import CONFIG_NAME, PeftType, TaskType\n  File \"/usr/local/lib/python3.10/dist-packages/peft/utils/__init__.py\", line 22, in <module>\n    from .utils import CONFIG_NAME, PeftType, TaskType\n  File \"/usr/local/lib/python3.10/dist-packages/peft/utils/__init__.py\", line 22, in <module>\n    from .loftq_utils import replace_lora_weights_loftq\n  File \"/usr/local/lib/python3.10/dist-packages/peft/utils/loftq_utils.py\", line 26, in <module>\n    from .loftq_utils import replace_lora_weights_loftq\n  File \"/usr/local/lib/python3.10/dist-packages/peft/utils/loftq_utils.py\", line 26, in <module>\n    from huggingface_hub.errors import HFValidationError, LocalEntryNotFoundError\nModuleNotFoundError: No module named 'huggingface_hub.errors'\n\nThe above exception was the direct cause of the following exception:\n\nTraceback (most recent call last):\n  File \"/kaggle/usr/lib/deepseek_tune/finetune_deepseekcoder.py\", line 11, in <module>\n    from huggingface_hub.errors import HFValidationError, LocalEntryNotFoundError\nModuleNotFoundError: No module named 'huggingface_hub.errors'\n\nThe above exception was the direct cause of the following exception:\n\nTraceback (most recent call last):\n  File \"/kaggle/usr/lib/deepseek_tune/finetune_deepseekcoder.py\", line 11, in <module>\n    from transformers import Trainer\n  File \"<frozen importlib._bootstrap>\", line 1075, in _handle_fromlist\n  File \"/usr/local/lib/python3.10/dist-packages/transformers/utils/import_utils.py\", line 1335, in __getattr__\n    module = self._get_module(self._class_to_module[name])\n  File \"/usr/local/lib/python3.10/dist-packages/transformers/utils/import_utils.py\", line 1347, in _get_module\n    raise RuntimeError(\nRuntimeError: Failed to import transformers.trainer because of the following error (look up to see its traceback):\nNo module named 'huggingface_hub.errors'\n    from transformers import Trainer\n  File \"<frozen importlib._bootstrap>\", line 1075, in _handle_fromlist\n  File \"/usr/local/lib/python3.10/dist-packages/transformers/utils/import_utils.py\", line 1335, in __getattr__\n    module = self._get_module(self._class_to_module[name])\n  File \"/usr/local/lib/python3.10/dist-packages/transformers/utils/import_utils.py\", line 1347, in _get_module\n    raise RuntimeError(\nRuntimeError: Failed to import transformers.trainer because of the following error (look up to see its traceback):\nNo module named 'huggingface_hub.errors'\n[2025-02-19 03:35:38,030] [INFO] [launch.py:315:sigkill_handler] Killing subprocess 200\n[2025-02-19 03:35:38,030] [INFO] [launch.py:315:sigkill_handler] Killing subprocess 201\n[2025-02-19 03:35:38,031] [ERROR] [launch.py:321:sigkill_handler] ['/usr/bin/python3', '-u', '/kaggle/usr/lib/deepseek_tune/finetune_deepseekcoder.py', '--local_rank=1', '--model_name_or_path', '--data_path', '--output_dir', '--num_train_epochs', '3', '--model_max_length', '1024', '--per_device_train_batch_size', '16', '--per_device_eval_batch_size', '1', '--gradient_accumulation_steps', '4', '--evaluation_strategy', 'no', '--save_strategy', 'steps', '--save_steps', '100', '--save_total_limit', '100', '--learning_rate', '2e-5', '--warmup_steps', '10', '--logging_steps', '1', '--lr_scheduler_type', 'cosine', '--gradient_checkpointing', 'True', '--report_to', 'tensorboard', '--deepspeed', '/kaggle/input/deepspeed-config/ds_config_zero3.json', '--bf16', 'True'] exits with return code = 1\n","output_type":"stream"}],"execution_count":19}]}